pom.xml

```diff
@@ -206,6 +207,12 @@
                 <version>${zjsonpatch.version}</version>
                 <scope>test</scope>
             </dependency>
+            <dependency>
+                <groupId>io.sundr</groupId>
+                <artifactId>builder-annotations</artifactId>
+                <version>${sundr-builder-annotations.version}</version>
+                <scope>provided</scope>
```

Comment:
I don't remember if I've ever followed through fully on scopes in dependency management but doesn't this enforce that it only ever appears in provided scope? 

Speaking which what provides it?

Reply:
> If others think there is enough value in keeping code generation I'm not going to fight about it, but I have a really high bar for needing two independent systems.

we are not actually using lombok in kroxylicious project at the moment.  It is a dead dependency.  The junit ext is using it.  

> I don't remember if I've ever followed through fully on scopes in dependency management but doesn't this enforce that it only ever appears in provided scope?

Sundr builder generates code at compile time.

Reply:
> Sundr builder annotations does support composition fluency, so I tried switching to that. However, I hit problems when applying that to config.Configuration. The builder was somehow losing state. I failed to get to the bottom of what was going wrong. I was falling down the rabbit hole/losing sight of the real task in hand. I could also foresee more issues with getting the sundr builders to construct with our config specialisations (FilterConfig etc).
>I retreated, falling back to replacing the hand rolled builder methods of KroxyConfig that you see in the PR.

That's exactly the sort of reason I don't like annotation driven code generation... As mentioned earlier I'm not going to stand in the way of people replacing hand rolled builders but I'm skeptical of the value added by doing so.

>> I don't remember if I've ever followed through fully on scopes in dependency management but doesn't this enforce that it only ever appears in provided scope?

> Sundr builder generates code at compile time.


OH so you're forcing `provided` scope so that it doesn't appear in the final dependency graph?